Add the `access_request.delete` event

espadolini opened the pull request:
Fixes half of TEL-Q321-7.

Closes #7078.

espadolini committed:
Add the `access_request.delete` event

espadolini commented:
The web interface for the audit log will need to support the new event as well.

smallinsky commented on lib/auth/auth.go:
```diff
@@ -2439,6 +2439,28 @@ func (a *Server) CreateAccessRequest(ctx context.Context, req types.AccessReques
+			Impersonator: ClientImpersonator(ctx),
+		},
+		ResourceMetadata: apievents.ResourceMetadata{
+			Name: name,
```
All other field uses `RequestID:    params.RequestID` Can we align the `AccessRequestDelete` struct and use `RequestID` field instead of ResourceMetadata.Name for consistency ?

espadolini replied:
I agree with you here; some resource types always use `ResourceMetadata` for all events, while other types always use a custom field - no resource type uses a mix, so to be consistent we should do what we already do for the `AccessRequestCreate` event.

smallinsky commented on lib/events/codes.go:
```diff
@@ -418,6 +418,8 @@ const (
 	AccessRequestUpdateCode = "T5001I"
 	// AccessRequestReviewCode is the access review application code.
 	AccessRequestReviewCode = "T5002I"
+	// AccessRequestCreateCode is the the access request deleted code.
```
```suggestion
	// AccessRequestDeleteCode is the access request deleted code.
```

espadolini committed:
fix typo

Co-authored-by: Marek Smoliński <marek@goteleport.com>

espadolini committed:
Use a custom RequestID field instead of ResourceMetadata in AccessRequestDelete

This is for consistency with AccessRequestCreate and other FooDelete
events for resources that don't use the Name field in Metadata to
identify the resource.

espadolini commented:
Greenlit by @russjones
